### PR TITLE
customized plugin manager to skipping running if no handles and registered

### PR DESCRIPTION
**Pull request by waynepeking348**
#### What type of PR is this?
bug fixes

#### What this PR does / why we need it:
customized plugin manager to skipping running if no handles and registered to avoid logging error logs too frequently

### CODE REVIEW

File: cmd/katalyst-agent/app/agent/context.go [hunk 1]
```diff
@@ -109,6 +111,31 @@ func (c *GenericContext) Run(ctx context.Context) {
...
+// customizedPluginManager requires that handlers must be added before
+// it really runs to avoid logging error logs too frequently
+type customizedPluginManager struct {
+	enabled *atomic.Bool
+	pluginmanager.PluginManager
+}
+
+// Run successes iff at-least-one handler has been registered
+func (p *customizedPluginManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
+	if p.enabled == nil || !p.enabled.Load() {
```
Comment (csfldf) [inline cmd/katalyst-agent/app/agent/context.go:123]: should we deal with the condition that PluginManager.Run is called before PluginManager.AddHandler? 

It's similar to [shared_informer adds listener after it's started](https://github.com/kubernetes/client-go/blob/master/tools/cache/shared_informer.go#L611).

File: cmd/katalyst-agent/app/agent/context.go [hunk 2]
```diff
@@ -109,6 +111,31 @@ func (c *GenericContext) Run(ctx context.Context) {
...
 }
 
+// customizedPluginManager requires that handlers must be added before
+// it really runs to avoid logging error logs too frequently
+type customizedPluginManager struct {
+	enabled *atomic.Bool
+	pluginmanager.PluginManager
+}
+
+// Run successes iff at-least-one handler has been registered
```
Comment (zzzzhhb) [inline cmd/katalyst-agent/app/agent/context.go:121]: "iff" typo?

File: cmd/katalyst-agent/app/agent/context.go (hunk 1 above)
Comment (luomingmeng) [reply cmd/katalyst-agent/app/agent/context.go:123]: in our case, the AddHandler must be called before Run, which is guaranteed by the agent framework.

**Review by luomingmeng — APPROVED** [review]

File: cmd/katalyst-agent/app/agent/context.go (hunk 2 above)
Comment (waynepeking348) [reply cmd/katalyst-agent/app/agent/context.go:122]: actually, iff = if and only if

**Review by csfldf — DISMISSED** [review]

File: cmd/katalyst-agent/app/agent/context.go [hunk 3]
```diff
@@ -109,6 +112,30 @@ func (c *GenericContext) Run(ctx context.Context) {
...
+	enabled atomic.Bool
+	pluginmanager.PluginManager
+}
+
+// Run successes iff at-least-one handler has been registered
+func (p *customizedPluginManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
+	for range time.Tick(time.Second) {
+		if p.enabled.Load() {
+			p.PluginManager.Run(sourcesReady, stopCh)
+			klog.Infof("started plugin-manager since handlers have been enabled")
```
Comment (zzzzhhb) [inline cmd/katalyst-agent/app/agent/context.go:127]: this log seems meaningless since the `PluginManager.Run` function only returns when it stops

**Review by luomingmeng — APPROVED** [review]

**Review by csfldf — APPROVED** [review]

**Review by zzzzhhb — APPROVED** [review]

**Pull request merged**